### ISSUE TITLE
false positives for `no-unused-vars` in `gjs` files, when using `<Component as |something|>`

### ISSUE THREAD

**Issue by c0rydoras** [ember-cli/eslint-plugin-ember#1991]
I am experiencing false positives for `no-unused-vars` in `.gjs` files

Example: 
```js
import UkCard from 'ember-uikit';

<template>
  <UkCard as |card|>
    <card.body>foo</card.body>
  </UkCard>
</template>
```
```
[lint:js]   /home[...]/ember-app/app/components/foo.gjs
[lint:js]   4:15  error  'card' is defined but never used  no-unused-vars
```
---
minimal reproduction: https://github.com/c0rydoras/gjs-linting-no-undef-reproduction
failing linting workflow: https://github.com/c0rydoras/gjs-linting-no-undef-reproduction/actions/runs/6798889824/job/18483951695

**Comment by NullVoxPopuli** [comment]
does anything improve if you remove this line?
https://github.com/c0rydoras/gjs-linting-no-undef-reproduction/blob/main/.eslintrc.js#L5C29-L5C29
![image](https://github.com/ember-cli/eslint-plugin-ember/assets/199018/41372994-2fd5-48d5-bcc7-346d65acbd24)
from: https://github.com/ember-cli/eslint-plugin-ember#gtsgjs

**Comment by c0rydoras** [comment]
sadly nothing changed: https://github.com/c0rydoras/gjs-linting-no-undef-reproduction/actions/runs/6802826029/job/18496779696

**Comment by patricklx** [comment]
It's an issue with gts parser... I look into it
